pkg/kubelet/userns/userns_manager.go

```diff
@@ -392,6 +393,13 @@ func (m *UsernsManager) GetOrCreateUserNamespaceMappings(pod *v1.Pod) (*runtimea
 			Mode: runtimeapi.NamespaceMode_NODE,
 		}, nil
 	}
+	handlerSupportsUserns, err := m.kl.HandlerSupportsUserNamespaces(runtimeHandler)
+	if err != nil {
+		return nil, err
+	}
+	if !handlerSupportsUserns {
+		return nil, fmt.Errorf("the handler %q does not support user namespaces", runtimeHandler)
+	}
```

Comment:
We want to do this check even if the feature gate is disabled. If the pod requests userns and the feature is disabled, PRR and mrunal said we should reject the pod.

Reply:
thanks, changed now.  Is the new version better?

Reply:
@giuseppe I don't think so :(

I've tested it locally in the following way:
1. Start a cluster with local-up-cluster and the userns feature gate
1. Create a pod, all was working fine
1. run "ps faux | grep kubelet" and copied the line
1. Kill the current kubelet, see that the output of local-up-cluster mentions the kubelet crashed
1. From the kubelet cmd line, see the config file and edit it to remove the userns feature gate
1. Start the kubelet again, by running the command we c&p
1. Create a pod with userns
1. The pod is not rejected (as it should), but it is create without userns